stl/inc/format

```diff
@@ -1353,22 +1353,71 @@ _NODISCARD constexpr bool _In_bounds(const _Ty _Value) {
     }
 }
 
+_NODISCARD inline int _Count_separators(size_t _Digits, const string_view _Groups) {
+    int _Separators = 0;
+    if (!_Groups.empty()) {
+        // Calculate the amount of separators that are going to be inserted based on the groupings of the locale.
+        auto _Group_it = _Groups.begin();
+        while (_Digits > static_cast<size_t>(*_Group_it)) {
+            _Digits -= static_cast<size_t>(*_Group_it);
+            ++_Separators;
+            if (_Group_it + 1 != _Groups.end()) {
+                ++_Group_it;
+            }
+        }
+    }
+    return _Separators;
+}
+
+template <class _CharT, class _OutputIt>
+_OutputIt _Write_separated_integer(const char* _Start, const char* const _End, const string_view _Groups,

```

Comment:
`_NODISCARD`? (I'm surprised that all of these `_Write_meow` functions aren't `_NODISCARD`.)

Reply:
there's an outstanding issue to nodiscardify a bunch of stuff.

Reply:
I'll let this decision be made then, but to me you don't _always_ care how much the iterator was advanced. I suppose given the nature of the formatters, we always do care, so we probably should `_NODISCARD` them, but I think we can do it later as a whole audit.